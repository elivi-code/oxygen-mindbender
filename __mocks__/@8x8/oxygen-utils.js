export * from '@8x8/oxygen-utils';

const useFontSize = jest.fn(() => 16);

export { useFontSize };
