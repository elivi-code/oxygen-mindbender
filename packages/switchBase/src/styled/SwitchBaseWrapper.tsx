import styled from 'styled-components';

export interface Props {
  isDisabled?: boolean;
  isFocused?: boolean;
  hasLabel?: boolean;
  isToggleButton?: boolean;
}

const SwitchBaseWrapper = styled.div<Props>`
  position: relative;
  display: flex;

  label {
    color: ${({ isDisabled, theme }) =>
      isDisabled ? theme.labelDisabledColor : theme.labelColor};
    cursor: ${({ isDisabled }) => isDisabled && 'not-allowed'};
    border-radius: 4px;
    align-items: flex-start;
    padding: ${({ isToggleButton }) =>
      isToggleButton ? '5px 3px' : '2px 2px 2px 0'};

    ${({ hasLabel, isFocused, theme, isToggleButton }) =>
      ((hasLabel && isFocused) || (isFocused && isToggleButton)) &&
      `
      box-shadow: 0 0 0 2px inset ${theme.focusColor};
    `}
  }
`;

export default SwitchBaseWrapper;
