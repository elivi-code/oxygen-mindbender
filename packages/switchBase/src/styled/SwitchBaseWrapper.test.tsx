import React from 'react';
import renderer from 'react-test-renderer';
import { DefaultTheme, ThemeProvider } from 'styled-components';

import { checkbox as theme } from '@8x8/oxygen-constants';
import SwitchBaseWrapper, { Props } from './SwitchBaseWrapper';

describe('<SwitchBaseWrapper>', () => {
  let wrapper;

  function setup(props: Props = { isDisabled: false }) {
    return renderer
      .create(
        <ThemeProvider theme={theme as DefaultTheme}>
          <SwitchBaseWrapper {...props} />
        </ThemeProvider>,
      )
      .toJSON();
  }

  it('should render correctly with default props', () => {
    wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when is disabled', () => {
    wrapper = setup({ isDisabled: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a focus ring when isFocused and hasLabel is set to true', () => {
    wrapper = setup({ isFocused: true, hasLabel: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a focus ring when isFocused and isToggleButton is set to true and hasLabel is false', () => {
    wrapper = setup({ isFocused: true, hasLabel: false, isToggleButton: true });
    expect(wrapper).toMatchSnapshot();
  });
});
