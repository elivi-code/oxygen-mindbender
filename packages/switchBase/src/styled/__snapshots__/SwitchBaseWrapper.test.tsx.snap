// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SwitchBaseWrapper> should render a focus ring when isFocused and hasLabel is set to true 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 label {
  color: #1C2025;
  border-radius: 4px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 2px 2px 2px 0;
  box-shadow: 0 0 0 2px inset #0056E0;
}

<div
  className="c0"
/>
`;

exports[`<SwitchBaseWrapper> should render a focus ring when isFocused and isToggleButton is set to true and hasLabel is false 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 label {
  color: #1C2025;
  border-radius: 4px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 5px 3px;
  box-shadow: 0 0 0 2px inset #0056E0;
}

<div
  className="c0"
/>
`;

exports[`<SwitchBaseWrapper> should render correctly when is disabled 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 label {
  color: rgba(28,32,37,0.5);
  cursor: not-allowed;
  border-radius: 4px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 2px 2px 2px 0;
}

<div
  className="c0"
/>
`;

exports[`<SwitchBaseWrapper> should render correctly with default props 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 label {
  color: #1C2025;
  border-radius: 4px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 2px 2px 2px 0;
}

<div
  className="c0"
/>
`;
